### PR TITLE
Add commit language option

### LINKED ISSUE #26982: Option to use a different language for commit messages than UI
### Feature Description

When performing actions through the web UI (Adding files, editing them or deleting them) without defining a commit message will Gitea default to a default message, which is used for the commit.
Said message is using the language you're viewing the UI in, so updating a file while using german would display `<filename> aktualisiert` as the commit message.

This at first may not be a big problem, but it can start to confuse people if you have your content in english, but prefer to use a different language for the UI.
This situation could confuse people who visit your repository and then see commit messages in a foreigin language they don't understand. And using a translator just to figure out that a commit means `<filename> updated` seems a bit much to ask for.

So, an option to set a different language than what the UI is set to should be introduced here.
By default should it be set to `(Use UI language)` meaning if you switch the language, the commit message language will then change too for future commits.

I would apreciate if this could be considered.

(Also, yes I am well aware that I could just put a commit message myself, but what is the purpose of these default messages then if not to safe time?)

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by Paul-Stern**
Add commit language option.
Fixes: https://github.com/go-gitea/gitea/issues/26982
Add an "app.ini" setting.
To get some details, please, read commit log message.
If any changes or feedback are needed, please, let me know.

### CODE REVIEW

File: modules/setting/ui.go [hunk 1]
```diff
@@ -64,6 +64,8 @@ var UI = struct {
 		Description string
 		Keywords    string
 	} `ini:"ui.meta"`
+
+	CommitLanguage string `ini:"ui.commit_language"`
```
Comment (6543) [inline modules/setting/ui.go:68]: if it'sjust a single string and not a subsection you dont have to do it this way

Comment (Paul-Stern) [reply modules/setting/ui.go:68]: I just tried to do this the way it was suggested in https://github.com/go-gitea/gitea/issues/26982#issuecomment-1712509924 . Could you, please, be more specific and tell me what exactly is not suitable so I could consider and change it?

Comment (Paul-Stern) [reply modules/setting/ui.go:68]: Should it be just as the following?
```go
	CommitLanguage string `ini:"COMMIT_LANGUAGE"`
```

**Comment by Paul-Stern** [comment]
> I would prefer to keep things simple: always use English for commit messages.
> 
> The reasons are:
> 
>     1. It's quite common to use English as commit message.
> 
>     2. End users could always input their message in their language: either by git client, or on the web UI, the default commit message is seldom useful.
> 
>     3. The translations are done by Crowdin, it's difficult to guarantee the correctness, eg: in some languages the translated commit message could be very wrong.
> 
> 
> So, I prefer to always use the English commit message on the UI as the default message.

@wxiaoguang I started to implement it this way because the corresponding issue has label "proposal/accepted". I didn't think it was still in discussion.

**Comment by Paul-Stern** [comment]
So should anything be done in this branch so that the PR could be merged? If yes, what kind of changes are expected?

**Comment by Paul-Stern** [comment]
@6543 I changed the CommitLanguage field in https://github.com/Paul-Stern/gitea/commit/fa764dcd65e0cc5bb176d4c5dd1062f994653871 . Does it look alright? Shall I introduce this change to current branch? Is there anything to be done before this PR can be merged?

**Comment by lunny** [comment]
> > I would prefer to keep things simple: always use English for commit messages.
> > The reasons are:
> > ```
> > 1. It's quite common to use English as commit message.
> > 
> > 2. End users could always input their message in their language: either by git client, or on the web UI, the default commit message is seldom useful.
> > 
> > 3. The translations are done by Crowdin, it's difficult to guarantee the correctness, eg: in some languages the translated commit message could be very wrong.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > So, I prefer to always use the English commit message on the UI as the default message.
> 
> @wxiaoguang I started to implement it this way because the corresponding issue has label "proposal/accepted". I didn't think it was still in discussion.

I'm very sorry I think we(maintainers) haven't have a consistent opionion for the issue before. But as I can see, most of maintainers would like to keep it as English-only now. I will close this for now. But anyway, thank you for your contribution. You are still welcome to send new contribution and I think we need more discussion next time.

**Pull request closed without merging**